### PR TITLE
docs: add note for environment variables on Vercel deployment

### PR DESCRIPTION
**Pull request by smaeda-ks**
Adding notes about how env vars get handled in Vercel deployments, specifically for Edge Runtime.

internal ref: https://vercel.slack.com/archives/C0676QZBWKS/p1701328878214989?thread_ts=1701130706.927319&cid=C0676QZBWKS

### CODE REVIEW

File: docs/02-app/01-building-your-application/07-configuring/03-environment-variables.mdx [hunk 1]
```diff
@@ -183,6 +183,8 @@ If you've configured [Development Environment Variables](https://vercel.com/docs
 vercel env pull .env.local
 ```
 
+> **Good to know**: When deploying your Next.js application to [Vercel](https://vercel.com), your environmet variables in `.env*` files will not be made available to Edge Runtime, unless their name are prefided with `NEXT_PUBLIC_`. We strognly recommend managing your environment variables in [Project Settings](https://vercel.com/docs/projects/environment-variables?utm_medium=docs&utm_source=next-site&utm_campaign=next-website) instead.
```
Comment (ckopecky) [inline docs/02-app/01-building-your-application/07-configuring/03-environment-variables.mdx:186]: **Good to know**: When deploying your Next.js application to [Vercel](https://vercel.com), your environment variables in `.env*` files will not be made available to Edge Runtime, unless their name are prefixed with `NEXT_PUBLIC_`. We strongly recommend managing your environment variables in [Project Settings](https://vercel.com/docs/projects/environment-variables?utm_medium=docs&utm_source=next-site&utm_campaign=next-website) instead.

Comment (ckopecky) [reply docs/02-app/01-building-your-application/07-configuring/03-environment-variables.mdx:186]: Just corrected spelling :)

Comment (smaeda-ks) [reply docs/02-app/01-building-your-application/07-configuring/03-environment-variables.mdx:186]: omg lol thanks for catching my typos!

**Review by delbaoliveira — APPROVED** [review]
Thank you @smaeda-ks 🙏🏼

**Review by balazsorban44 — APPROVED** [review]

**Pull request merged**